lib/galaxy/tools/parameters/basic.py

```diff
@@ -235,6 +234,17 @@ def build( cls, tool, param ):
         else:
             return parameter_types[param_type]( tool, param )
 
+    @classmethod
+    def parse_name(cls, input_source):
+        name = input_source.get("name", None)
+        if name is None:
+            argument = input_source.get("argument")
+            if argument:
+                name = argument.lstrip("-")
+            else:
```

Comment:
I'd put it the other way 'round.  You can always refer to a parameter as
--{name} (as an argument) if argument isn't specified.  Name is always
required.

On Fri, Mar 6, 2015 at 4:08 PM John Chilton <notifications@github.com>
wrote:

> In lib/galaxy/tools/parameters/basic.py
> <https://github.com/galaxyproject/galaxy/pull/8#discussion_r25978130>:
>
> > @@ -235,6 +234,17 @@ def build( cls, tool, param ):
> >          else:
> >              return parameter_types[param_type]( tool, param )
> >
> > +    @classmethod
> > +    def parse_name(cls, input_source):
> > +        name = input_source.get("name", None)
> > +        if name is None:
> > +            argument = input_source.get("argument")
> > +            if argument:
> > +                name = argument.lstrip("-")
> > +            else:
>
> It shouldn't modify argument - to me it makes sense that if you specify
> the argument as --input - the name can safely be inferred to be input.
> This provides some actual convenience to tool authors annotating things
> this way - argument will still be --input.
>
> Put another way...
>
> <param name="input" argument="--input"...
>
> feels redundant right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/galaxyproject/galaxy/pull/8/files#r25978130>.
>